patches/server/1060-Implement-preliminary-API-for-working-with-new-mob-A.patch

```diff
@@ -0,0 +1,861 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Janet Blackquill <uhhadd@gmail.com>
+Date: Fri, 26 Apr 2024 23:46:35 -0400
+Subject: [PATCH] Implement preliminary API for working with 'new' mob AI
+ system
+
+
+diff --git a/src/main/java/io/papermc/paper/entity/ai/PaperActivities.java b/src/main/java/io/papermc/paper/entity/ai/PaperActivities.java
+new file mode 100644
+index 0000000000000000000000000000000000000000..33752f9b35676d8794ca320f7c9b8aa727aff2d6
+--- /dev/null
++++ b/src/main/java/io/papermc/paper/entity/ai/PaperActivities.java
+@@ -0,0 +1,111 @@
++package io.papermc.paper.entity.ai;
++
++import org.jetbrains.annotations.NotNull;
++import net.minecraft.world.entity.schedule.Activity;
++
++public class PaperActivities implements io.papermc.paper.entity.ai.Activities {
++    @Override
++    public @NotNull io.papermc.paper.entity.ai.Activity getAdmireItem() {
```

Comment:
I'm totally against exposing that via method that way.
I personally prefer a key based approach, where you can use a paper api-generator module to generate these keys automatically. You can even generate the mapping for the keys automatically.


```suggestion
+    public @NotNull io.papermc.paper.entity.ai.Activity get(PaperActivity activity) {
```

Reply:
removed